src/main/java/org/jabref/gui/groups/GroupSidePane.java

```diff
@@ -20,15 +25,43 @@
 public class GroupSidePane extends SidePaneComponent {
 
     private final JabRefPreferences preferences;
+    private final DialogService dialogService;
+    private final Button intersectionUnionToggle = IconTheme.JabRefIcons.WWW.asButton();
+    private final Tooltip toggleUnion = new Tooltip(Localization.lang("Toogle union"));
+    private final Tooltip toggleIntersectoin = new Tooltip(Localization.lang("Toogle intersection"));
 
-    public GroupSidePane(SidePaneManager manager, JabRefPreferences preferences) {
+    public GroupSidePane(SidePaneManager manager, JabRefPreferences preferences, DialogService dialogService) {
         super(manager, IconTheme.JabRefIcons.TOGGLE_GROUPS, Localization.lang("Groups"));
         this.preferences = preferences;
+        this.dialogService = dialogService;
+    }
+
+    @Override
+    protected Optional<Node> getAddtionalHeaderButtons() {
+        intersectionUnionToggle.setOnAction(event -> toggleUnionIntersection());
+        return Optional.of(intersectionUnionToggle);
+    }
+
+    private Node getUnionIntersectionGraphic() {
+        return preferences.getGroupViewMode().getIcon().getGraphicNode();
```

Comment:
Personally, I would extract these methods to a new class `GroupViewModeViewModel` which gives you the graphic and the tooltip (constructor accepts the mode). In this way, you clearly separate everything ui-related from the actual model class.